qos-monitor/src/main/java/eu/arrowhead/core/qos/service/ping/monitor/PingEventCollectorTask.java

```diff
@@ -0,0 +1,153 @@
+package eu.arrowhead.core.qos.service.ping.monitor;
+
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.LinkedBlockingQueue;
+
+import javax.annotation.Resource;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import eu.arrowhead.common.dto.shared.EventDTO;
+import eu.arrowhead.common.dto.shared.QosMonitorEventType;
+import eu.arrowhead.common.exception.ArrowheadException;
+import eu.arrowhead.common.exception.InvalidParameterException;
+import eu.arrowhead.core.qos.QosMonitorConstants;
+import eu.arrowhead.core.qos.dto.event.EventDTOConverter;
+import eu.arrowhead.core.qos.dto.event.monitoringevents.FinishedMonitoringMeasurementEventDTO;
+import eu.arrowhead.core.qos.dto.event.monitoringevents.InterruptedMonitoringMeasurementEventDTO;
+import eu.arrowhead.core.qos.dto.event.monitoringevents.MeasurementMonitoringEvent;
+import eu.arrowhead.core.qos.dto.event.monitoringevents.ReceivedMonitoringRequestEventDTO;
+import eu.arrowhead.core.qos.dto.event.monitoringevents.StartedMonitoringMeasurementEventDTO;
+
+public class PingEventCollectorTask implements Runnable {
+
+	//=================================================================================================
+	// members
+
+	private static final String NOT_SUPPORTED_EVENT_TYPE = " is not a supported event type. ";
+	private static final String REPLACEING_MEASURMENT_EVENT = " - measurment , duplicate event. Overwriting : ";
+
+	private final long clearingInterval = 1000 * 60 * 10;
+
+	private boolean interrupted = false;
+	private long lastBufferCleanAt;
+
+	@Resource(name = QosMonitorConstants.EVENT_QUEUE)
+	private LinkedBlockingQueue<EventDTO> eventQueue;
+
+	@Resource(name = QosMonitorConstants.EVENT_BUFFER)
+	private ConcurrentHashMap<UUID, PingEventBufferElement> eventBuffer;
+
+	@Autowired
+	private PingEventBufferCleaner bufferCleaner;
+
+	private final Logger logger = LogManager.getLogger(PingEventCollectorTask.class);
+
+	//=================================================================================================
+	// methods
+
+	//-------------------------------------------------------------------------------------------------	
+	@Override
+	public void run() {
+		logger.debug("PingEventCollectorTask run started...");
+
+		interrupted = Thread.currentThread().isInterrupted();
+		clearBuffer();
+
+		while (!interrupted) {
+			logger.debug("PingEventCollectorTask run loop started...");
+
+			try {
+				putEventToBuffer(eventQueue.take());
+
+				if (lastBufferCleanAt + clearingInterval < System.currentTimeMillis()) {
+
+					lastBufferCleanAt = System.currentTimeMillis();
```

Comment:
maybe changing the order here: first the work, then the timestamp

Reply:
fixed